request_test.go

```diff
@@ -504,82 +504,212 @@ func TestRequest_URLConcatenate(t *testing.T) {
 
 	reporter := NewAssertReporter(t)
```

Comment:
Let's use reporter creation into subtest, so that it will use local `t, and subtests would be more isolated. Client also can be moved there.

Reply:
Same for TestRequest_URLOverwrite